hips/hip-0013.md

```diff
@@ -0,0 +1,94 @@
+
+The [cache provided for the Helm client](https://github.com/rancher/sandbox/gofilecache) is based on the Go build cache and uses the file system. It uses the first two characters of the hash as the top directory followed by the rest of the hash for the second level directory. This is similar to the way in with Git stores objects on disk. Both the chart and provenance caches will live within the cache directory Helm already uses.
+
+The Helm Client would have two new environment variables and flags to specify the locations for the provenance files and chart archive cache. These would follow the same format as the current environment variable use for individual caching locations.
+
+## Backwards compatibility
+
+Helm charts are untouched in their structure. There is no impact to the charts themselves.
+
+Index files, where digests are looked up, remain untouched. There is no impact to the index files.
+
+The existing Helm cache directory is used for the cache storage. This location needs to exist for storing index files locally. There is no impact to the cache location.
+
+Those who use the `ChartDownloader` from the Helm SDK will have two new properties to define when instantiating it.
+
+## Security implications
+
+If a chart archive or a provenance file is in the cache it will be used instead of being downloaded. This means that a malicious user could place a chart archive in the cache in the location where the good one should be. When Helm looks up the chart listed in the repository it would retrieve the the malicious chart from the cache instead.
+
+For this to work the malicious user would need access to the system Helm is running on with write access to the cache. The current cache directory does not provide write access to those who are not the user or a system admin. That would mean the malicious user would need access as the user or a system admin to exploit this issue.
+
+To further mitigate the issue, `ChartDownloader` will check that the archive conforms to the digest that is being requested. If it does not the `ChartDownloader` will retrieve it from the source.
+
+This does mean that if the source chart archive does not have the same digest the one listed in the index it will be retrieved every time and bypass the cache.
+
+## How to teach this
+
+The Helm client will have a flag to bypass the cache. This will be in the client documentation alongside the other flags.
```

Comment:
Again for posterity, can we give that flag a name so this document can be self-describing?